src/cfnlint/helpers.py

```diff
@@ -171,6 +171,39 @@
     'AWS::Redshift::Cluster'
 ]
 
+VALID_PARAMETER_TYPES_SINGLE = [
+    'AWS::EC2::AvailabilityZone::Name',
+    'AWS::EC2::Image::Id',
+    'AWS::EC2::Instance::Id',
+    'AWS::EC2::KeyPair::KeyName',
+    'AWS::EC2::SecurityGroup::GroupName',
+    'AWS::EC2::SecurityGroup::Id',
+    'AWS::EC2::Subnet::Id',
+    'AWS::EC2::VPC::Id',
+    'AWS::EC2::Volume::Id',
+    'AWS::Route53::HostedZone::Id',
+    'AWS::SSM::Parameter::Name',
+    'Number',
+    'String',
+]
+
+VALID_PARAMETER_TYPES_LIST = [
+    'CommaDelimitedList',
+    'List<AWS::EC2::AvailabilityZone::Name>',
+    'List<AWS::EC2::Image::Id>',
+    'List<AWS::EC2::Instance::Id>',
+    'List<AWS::EC2::SecurityGroup::GroupName>',
+    'List<AWS::EC2::SecurityGroup::Id>',
+    'List<AWS::EC2::Subnet::Id>',
+    'List<AWS::EC2::VPC::Id>',
+    'List<AWS::EC2::Volume::Id>',
+    'List<AWS::Route53::HostedZone::Id>',
+    'List<Number>',
+    'List<String>',
```

Comment:
actually not seeing `List<String>` [documented](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/parameters-section-structure.html#parameters-section-structure-properties), so just wanna double check that's valid

Reply:
it is valid (I think).  I thought I had tested this before but left it off because it isn't documented any where.